### DISCUSSION TITLE
Crash in iOS 14.3 when using ForEachStore in TabView in tag 0.38.3

### DISCUSSION THREAD

**Discussion by ACFancy** [pointfreeco/swift-composable-architecture#1724] [Q&A] [answered]
## Crash in iOS 14.3 when using ForEachStore in TabView
- any idea for solved this crash problem ? 
``` swift
struct AState: Equatable, Identifiable {
  let uuid: UUID()
  var id: String { uuid.uuidString  }
}


struct BState: Equatable {
   var aStates: IdentifiedArrayOf<AState> = []
   @Bindable var selectedIndex: String = ""
   ...
}

enum  BAction: BindableAction {
    case aAction(String, AAction)
}

struct BView: View {
     let store: <BState, BAction>
     @ObservedObject private var viewStore: ViewStore<BState, BAction>
     ...
     var body: some View { 
            TabView(selectIndex: viewStore.binding(\.$selectedIndex)) {
                    ForEachState(store.scope(\.aStates, action: /BAction.aAction)) {
                      }
            }
      }
}

```

**Comment by mbrandonw** [comment]
Hi @ACFancy, there isn't enough info here for us to see why there might be a crash. Can you provide a sample project that reproduces the problem?

**Reply by ACFancy** [reply]
@mbrandonw SwiftUI TabView page style bug for iOS 14.xx, I using UIKit in SwiftUI to fixed that.

**Comment by erikbdev** [comment]
> ```
>    var aStates: IdentifiedArrayOf<AState> = []
> ```

This most likely has nothing to do with Composable Architecture. Instead, this crash will most likely have to do with the `aStates` containing no elements by the time TabView draws, which TabView requires at least one element or else it crashes.

**Reply by ACFancy** [reply]
no elements above iOS 15.xx is ok, no crash, so, SwiftUI TabView page style bug in iOS 14.xxx

**Comment by ACFancy** [comment, marked as answer]
SwiftUI TabView page style bug for iOS 14.xx, I using UIKit in SwiftUI to fixed that.
